.github/bin/cmp_py_versions

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+# Copyright (C) British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Check if two PEP 440 compliant Python package version numbers are equal.
+
+Usage:
+    $ cmp_py_versions <version_number1> <version_number2>
+
+    You can also check if a single version number is valid by passing it twice.
+"""
+
+import sys
+import packaging.version
+
+ver1, ver2 = [packaging.version.Version(i) for i in sys.argv[1:3]]
+
+if ver1 != ver2:
+    print(f'{ver1} not equal to {ver2}')
```

Comment:
Better to `exit(1)` rather than relying on stdout to communicate failure.

Not a biggie for this usage but worth noting that stdout/err can be quite fragile interfaces due to buffering (see the env var `PYTHONUNBUFFERED`), control characters, shell startup files, system messages and all sorts of lovely things.

Reply:
Only problem is that makes it more difficult to handle in the node script step of the workflow https://github.com/metomi/isodatetime/blob/2e5b4c474ad909079e4bcd4fa6b168bdcb70f354/.github/workflows/create_release_pr.yml#L94-L100